### ISSUE TITLE
Many points outside of the 95ci

### ISSUE THREAD

**Issue by Pearcekieser** [facebook/Ax#260]
Hi, I have been using ax for a few weeks and am really enjoying it.  Not sure if this is the right place to ask this question, but ..

I am using the service API to optimize a solution that take in one variable and outputs a result (ex. (x1) -> y1).  When I visualize the model with `render(plot_slice(exp.generation_strategy.model, "x1", "y1"))`.  The line doesn't trace the data points well and many data points are outside of the 95% ci.  This concerns me because if the ci does not approximate the function it wont guess points around there while optimizing (see the zoomed in figure below):

![image](https://user-images.githubusercontent.com/5055971/75737284-3322aa80-5cb4-11ea-84f4-087689f97225.png)

Zoomed in:

![image](https://user-images.githubusercontent.com/5055971/75737936-ce684f80-5cb5-11ea-9b0e-a58ab39f8792.png)

For the input I set x1:
```
parameters=[
                {
                    "name": "x1",
                    "type": "range",
                    "bounds": [1e-100, 1e100],
                    "value_type": "float",  # Optional, defaults to inference from type of "bounds".
                    "log_scale": True,  
                }
            ]
```

I pre-loaded the ax_client with existing results from x1 in [0.01, 0.1, 0.2, 0.4, 0.6, 1.0, 2.0, 4.0, 6.0, 10.0, 100.0, 1000.0, 10000.0], and the corresponding y1's.  Then I ran the simulation for 24 epochs, each epoch I ran 4 experiments in parallel with x1 that I got from ax_client.get_next_trial().  After those finished, I used ax_client.complete_trial() give the results back to ax_client, (note: i set the variance to 0.0 on the reported result as the simulation does not give that insight).  

I'd really appreciate some insight on why the ax client is having difficulty approximating this function.  And some advice on how to get things working better.  Thank you so much for your help!

**Comment by ldworkin** [comment]
Hi @Pearcekieser ! We'll definitely take a look at this. Can you provide a fully reproducible code example, so we can investigate? We'll need to see what function you're working with, for instance.

**Comment by Pearcekieser** [comment]
The results are based off of a simulation, so a cannot easily provide fully reproducible code example.  I am attaching a zip file containing some files of interest:
- `ax_test.ipynb` - this is a notebook that was used to generate these results.
- `W3_blind_bo.json` - the output from `ax_client.save_to_json_file()` of the current stuff in `ax_test.ipynb` (exhibits the problem above to a lesser degree)
- `W4_aware_bo.json` - the output from `ax_client.save_to_json_file()` of the experiment that created the graphs in the issue

the Runner uses the following code to return the results (this is used for both the newly run simulations and the lazy loaded results):
```python
def run_exp(self, trial_index_to_alphas):
        
        # ... run these experiments on the simulator

        index_to_results = {}
        for index, alpha in trial_index_to_alphas.items():
            
            # ... look up the given results for that alpha

            results = {'workload': [workload, 0.0], 'mean': [mean, 0.0], 'median': [p50, 0.0], 'p75': [p75, 0.0], 'p90': [p90, 0.0], 'p99': [p99, 0.0],
                'sd_mean': [sd_mean, 0.0], 'sd_median': [sd_p50, 0.0], 'sd_p75': [sd_p75, 0.0], 'sd_p90': [sd_p90, 0.0], 'sd_p99': [sd_p99, 0.0]}

            index_to_results[index] = results
        
        return index_to_results
```

My guess is it is a problem with how I am returning the results, or with how ax is handling getting results from the prepossessed runs and/or parallel runs.  

[sample.zip](https://github.com/facebook/Ax/files/4284424/sample.zip)

Thank you for the help!

**Comment by stevemandala** [comment]
Hey @Pearcekieser! We had a chance to do some preliminary investigation of the W4_aware_bo experiment, looking at the observed data and corresponding predictions from the posterior model, it appears that the model does have a good fit across all observed points (measured data well within the 95% CI). We believe there is an issue with the plotting library  (specifically `plot_slice()`) how it is handling the log scale param. Will let you know more as we investigate this.

**Comment by stevemandala** [comment]
Quick follow up, it does appear to be a shortcoming of the `plot_slice()` method. In particular, the library isn't generating a representative set of points around the region `alpha=1`. As a result, the plotted curve and CI aren't properly showing up in the region you shared. Ideally, the library should plot at the existing observed points or more points around these dense areas.

As a mitigation, `plot_slice()` has a `density` argument which you can specify how many points to sample from the range and plot to generate the curve. The default is 50, but setting it to 1000 or more, I was able to get more representative plot.

![ax_plot](https://user-images.githubusercontent.com/9901641/76011996-8890ce80-5ee3-11ea-8835-19194b7b172f.JPG)

**Comment by Pearcekieser** [comment]
Awesome thanks for the help @stevemandala ! That's a much more clear visualization.  

Do you know where I could learn more on how to configure the model? 
I would like for it to explore more points around that alpha = 4 area.  I could just code a new constraint and run the optimization again, but I am wondering if there is a way to configure the model that might better approximate the function.  

Thanks again for the help!

**Comment by stevemandala** [comment]
To my knowledge, there isn't a general way to force the EI searcher to explore a specific region of the search space, constraints are probably the best method here.

**Comment by ldworkin** [comment]
Marking this issue closed since increasing the density helped, and there isn't a problem with the underlying model fit. @Pearcekieser if you want to discuss the issue of forcing exploration in a different area further, you can open a separate issue. Thanks!
